subprojects/docs/src/docs/userguide/migrating_from_maven.adoc

```diff
@@ -0,0 +1,518 @@
+====
+
+As you can see, the Gradle configuration is often much shorter than the Maven equivalent.
+You also have a much more flexible execution model since we are not longer constrained by Maven's fixed phases.
+
+While migrating a project from Maven, don't forget about source sets.
+These often provide a more elegant solution for handling integration tests or generated sources than Maven can provide, so you should factor them into your migration plans.
+
+
+=== Ant goals
+
+Many Maven builds rely on the AntRun plugin to customize the build without the overhead of implementing a custom Maven plugin.
+Gradle has no equivalent plugin because Ant is a first-class citizen in Gradle builds, via the `ant` object.
+For example, you can use Ant's Echo task like this:
+
+.Invoking Ant tasks
+====
+include::sample[dir="userguide/mavenMigration/ant/groovy",files="build.gradle"]
+====
+
+Even Ant properties and filesets are supported natively.
+To learn more, see <<ant#ant,Using Ant from Gradle>>.
+
+
+== Understanding which plugins you don't need
+
+It's worth remembering that Gradle builds are typically easier to extend and customize than Maven.
+In this context, that means you may not need a Gradle plugin to replace a Maven one.
+For example, the Maven Enforcer plugin allows you to control dependency versions and environmental factors, but these things can easily be configured in a normal Gradle build script.
+
+
+== Dealing with uncommon and custom plugins
+
+You may come across Maven plugins that have no counterpart in Gradle, particularly if you or someone in your organisation has written a custom plugin.
+Such cases rely on you understanding how Gradle (and potentially Maven) works, because you will usually have to write your own plugin.
+
+For the purposes of migration, there are two key types of Maven plugin:
+
+* Those that use the Maven project object.
+* Those that don't.
+
+Why is this important? Because if you use one of the latter, you can trivially reimplement it as a Gradle task.
+Simply define task inputs and outputs to correspond to the mojo parameters and convert the execution logic into a task action.
+
+If a plugin depends on the Maven project, then you will have to rewrite it.
+Don't start by considering how the Maven plugin works, but look at what problem it is trying to solve.
+Then try to work out how to solve that problem in Gradle.
+You'll probably find that the two build models are different enough that "transcribing" Maven plugin code into a Gradle plugin just won't be effective.
+On the plus side, the plugin is likely to be much easier to write than the original Maven one because Gradle has a much richer build model.
+
+If you do need to implement custom logic, either via build files or plugins, then be sure to familiarize yourself with Gradle's link:{groovyDslPath}/[Groovy DSL Reference], which provides comprehensive documentation on the API that you'll be working with.
```

Comment:
I'd suggest we link to plugin development guides instead of the Groovy DSL reference, because it's a much better starting place.